src/Validation/Validator.php

```diff
@@ -525,6 +527,14 @@ protected function _fieldIsEmpty($data)
         if (empty($data) && $data !== '0' && $data !== false && $data !== 0 && $data !== 0.0) {
             return true;
         }
+        $isArray = is_array($data);
+        if ($isArray && (isset($data['year']) || isset($data['hour']))) {
+            $value = implode('', $data);
+            return strlen($value) === 0;
+        }
+        if ($isArray && isset($data['name'], $data['type'], $data['tmp_name'], $data['error'])) {
+            return (int)$data['error'] === UPLOAD_ERR_NO_FILE;
```

Comment:
I wonder if we should check for `!= UPLOAD_ERROR_OK` instead. The reason is suppose user does upload a file and error `UPLOAD_ERR_INI_SIZE` occurs the user would get an error message saying "field cannot be empty" which can be confusing to him. I personally add extra validation rules to check for other potential upload errors and show more appropriate message to user.